packages/core/src/browser/xhrProxy.ts

```diff
@@ -91,33 +93,47 @@ function proxyXhr() {
         const xhrContext = this._datadog_xhr as XhrStartContext & Partial<XhrCompleteContext>
         xhrContext.startTime = relativeNow()
 
-        const originalOnreadystatechange = this.onreadystatechange
+        const completeContext = () => {
+          if (xhrContext.duration !== undefined) {
+            return
+          }
+          xhrContext.duration = elapsed(xhrContext.startTime, relativeNow())
+          xhrContext.response = this.response as string | undefined
+          xhrContext.status = this.status
+          if (xhrContext.isAborted === undefined) {
+            xhrContext.isAborted = false
+          }
+        }
 
+        const originalOnreadystatechange = this.onreadystatechange
         this.onreadystatechange = function () {
           if (this.readyState === XMLHttpRequest.DONE) {
-            callMonitored(reportXhr)
+            // Try to complete the context as soon as possible, because the XHR may be mutated by
```

Comment:
it seems that it was already the case but the onreadystate override strategy could lead to unexpected behaviors for cases where the xhr is "recycled":
```
var xhr = new XMLHttpRequest()
xhr.open('GET', 'http://localhost:8080')
xhr.send()
xhr.open('GET', 'http://localhost:8080')
xhr.send()
```

it could be handled in a different task though. 
wdyt?

Reply:
I reverted most of the code to the previous implementation, so it shouldn't introduce any regression at this level. We shall investigate this further in a future task. I'll open a card about this.